server/plugin.go

```diff
@@ -49,21 +51,36 @@ func (p *Plugin) OnActivate() error {
 	teamSplit := split[0]
 	channelSplit := split[1]
 
-	team, err := p.API.GetTeamByName(teamSplit)
-	if err != nil {
-		return err
+	team, appErr := p.API.GetTeamByName(teamSplit)
+	if appErr != nil {
+		return appErr
 	}
 	p.TeamID = team.Id
 
-	user, err := p.API.GetUserByUsername(p.configuration.Username)
+	botID, err := p.Helpers.EnsureBot(&model.Bot{
+		Username:    "aws-sns",
+		DisplayName: "AWS SNS Plugin",
+		Description: "A bot account created by the plugin AWS SNS",
+	})
```

Comment:
A quick drive by review:
The whole code in https://github.com/mattermost/mattermost-plugin-aws-SNS/pull/32/files#diff-7e42e9629fece6c3b56011e7de76423bR70-R81 can be substituted by an option for `p.Helpers.EnsureBot`:
```suggestion
	botID, err := p.Helpers.EnsureBot(&model.Bot{
		Username:    "aws-sns",
		DisplayName: "AWS SNS Plugin",
		Description: "A bot account created by the plugin AWS SNS",
	},
	plugin.ProfileImagePath("assets/profile.png")
	)
```

Reply:
I agree with this suggestion. @vesari Out of curiosity, which plugin's code was your reference for this block?

Reply:
@mickmister , I think it was the GitLab plugin.

Reply:
@hanzei thanks, I applied your suggestion. Is there anything else I should change?